ocw/lib/provider.py

```diff
@@ -7,12 +7,22 @@
 import logging
 
 
+class ProviderAdapter(logging.LoggerAdapter):
+    """
+    ProviderAdapter provides the logger with a 'ns' key,
+    whose value in brackets is prepended to the log message.
+    """
+    def process(self, msg, kwargs):
+        return '[{}] {}'.format(self.extra['ns'], msg), kwargs
+
+
 class Provider:
 
     def __init__(self, namespace: str):
         self._namespace = namespace
         self.dry_run = PCWConfig.getBoolean('default/dry_run')
         self.logger = logging.getLogger(self.__module__)
```

Comment:
```suggestion
        self.logger = ProviderAdapter(logging.getLogger(self.__module__), {'ns': namespace}) 
```

no point to have two loggers let's have one

Reply:
please ignore this